torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -245,8 +247,11 @@ def wait_for_unshard(self):
         self._all_gather_result = None  # free unless saved in `all_gather_state`
 
     def _wait_all_gather_streams_on_event(self, event: torch.cuda.Event):
-        self.comm_ctx.all_gather_copy_in_stream.wait_event(event)
-        self.comm_ctx.all_gather_stream.wait_event(event)
+        # Calling `unshard` before lazy init means streams are not initialized
```

Comment:
In the end, I decided to just go with this `hasattr` check. Eagerly initializing CUDA streams in `FSDPCommContext` is painful because CUDA might not be available in some tests / requiring that might be strict. It is nice to be able to at least init with FSDP without CUDA (e.g. shard a meta-device model).

Reply:
It is also nice for typing checks that the attributes are always `torch.cuda.Stream`, not `Optional[torch.cuda.Stream]`.